mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2148,6 +2148,20 @@ where
             0
         };
 
+        // Remove transactions in the history_map that are not in the requested transaction list anymore
+        let requested_ids: HashSet<H256Json> = tx_ids.iter().map(|x| x.0).collect();
```

Comment:
This all can be rewritten to
```rust
history_map.retain(|hash, tx| tx_ids.contains(&(*hash, tx.block_height)));
```
In general, please check if your code can be simplified as above, especially if it contains several additional heap allocations (two new HashSet) and loops to perform a small task.